src/commcare_cloud/commands/ansible/service.py

```diff
@@ -1,328 +1,444 @@
+                    and matches(worker_num, worker_match):
+                processes_by_host[host].add(process_name)
+
+        processes_by_hosts = {}
+        # group hosts together so we do less calls to ansible
+        for grouper in groupby(processes_by_host.items(), key=lambda hp: hp[1]):
+            hosts = tuple(host_processes[0] for host_processes in grouper[1])
+            processes = grouper[0]
+            processes_by_hosts[hosts] = processes
+        return processes_by_hosts
+
+    def get_managed_services(self):
+        workers = get_celery_workers(self.environment)
+        return sorted({
+            '{}{}'.format(queue, ':{}'.format(worker_num) if worker_num > 1 else '')
+            for host, queue, worker_num, process_name in workers
+        })
+
+
+SERVICES = [
+    Postgresql,
+    Pgbouncer,
+    Nginx,
+    Couchdb,
+    RabbitMq,
+    Elasticsearch,
+    Redis,
+    Riakcs,
+    Kafka,
+    Webworker,
+    Formplayer,
+    Touchforms,
+    Celery,
+    CommCare,
+]
+
+SERVICE_NAMES = sorted([
+    service.name for service in SERVICES
+])
+
+SERVICES_BY_NAME = {
+    service.name: service for service in SERVICES
+}
+
+
+class Service(CommandBase):
+    command = 'service'
+    help = "Manage services. Use 'help' action for service details."
+
+    def make_parser(self):
+        self.parser.add_argument(
+            'services',
+            nargs="+",
+            choices=SERVICE_NAMES,
+            help="The services to run the command on"
+        )
+        self.parser.add_argument(
+            'action', choices=ACTIONS,
+            help="What action to take"
+        )
+        self.parser.add_argument('--limit', help=(
+            "Restrict the hosts to run the command on. Use 'help' action to list all option."
+        ))
+        self.parser.add_argument('--only', help=(
+            "Sub-service name to limit action to. Use 'help' action to list all option."
+        ))
 
     def run(self, args, unknown_args):
-        service_group = args.service_group
-        args.remote_user = 'ansible'
-        args.become = True
-        args.become_user = False
-        args.silence_warnings = True
-        if args.only:
-            self.ensure_permitted_only_options(service_group, args)
-        action = args.action
-        if action == "status":
-            exit_code = self.run_status_for_service_group(service_group, args, unknown_args)
-        else:
-            exit_code = self.perform_action(service_group, args, unknown_args)
-        return exit_code
+        environment = get_environment(args.env_name)
+
+        services = [
+            SERVICES_BY_NAME[name]
+            for name in args.services
+        ]
+
+        ansible_context = AnsibleContext(args)
+        exit_codes = []
+        for service_cls in services:
+            service = service_cls(environment, ansible_context)
+            exit_codes.append(service.run(args.action, args.limit))
+        return max(exit_codes)
```

Comment:
if exit_codes is `[0, -1]` this will return `0`, but I think the right answer is to return `-1`. Maybe only append non-zero statuses?